app/lib/package/model_properties.dart

```diff
@@ -41,7 +45,17 @@ class Pubspec {
 
   String get name => _inner.name;
 
-  String get version => _inner.version.toString();
+  String get nonCanonicalVersion => _inner.version.toString();
+  String get canonicalVersion {
+    if (_canonicalVersion == null) {
+      _canonicalVersion = canonicalizeVersion(nonCanonicalVersion);
+      if (_canonicalVersion == null) {
+        throw InvalidInputException.canonicalizeVersionError(
+            nonCanonicalVersion);
```

Comment:
This is not a good choice... If we load something from data-store it should never be in a state where we can't canonicalize it.

Reply:
Can we throw an `AssertionError` instead, and then have a check for this at upload-time?

Reply:
I think at that point it doesn't really matter which exception we throw (because it is some internal processing), but I can change this to throw `AssertionError`.